common/src/main/java/tachyon/security/UserInformation.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+
+/**
+ * This class wraps a JAAS Subject and provides methods to determine the user's name. It supports
+ * Windows, Unix, and Kerberos login modules.
+ */
+public class UserInformation {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private static final String OS_LOGIN_MODULE_NAME;
+  private static final String OS_PRINCIPAL_CLASS_NAME;
+  private static final boolean WINDOWS = System.getProperty("os.name").startsWith("Windows");
+  private static final boolean IS_64_BIT = System.getProperty("os.arch").contains("64");
+  private static final boolean AIX = System.getProperty("os.name").equals("AIX");
+  public static final String JAVA_VENDOR_NAME = System.getProperty("java.vendor");
```

Comment:
Do we have particular reasons to have `JAVA_VENDOR_NAME` and `IBM_JAVA` to be  `public`, while other members above are `private`?